packages/esm-patient-search-app/src/patient-search-result/patient-search-result.component.tsx

```diff
@@ -1,56 +1,48 @@
-import React from 'react';
-import dayjs from 'dayjs';
+import React, { useEffect, useState, useCallback } from 'react';
 import styles from './patient-search-result.scss';
-import { interpolateString, ExtensionSlot, useConfig, navigate } from '@openmrs/esm-framework';
+import { ExtensionSlot, useConfig, fetchCurrentPatient, interpolateString, navigate } from '@openmrs/esm-framework';
 import { SearchedPatient } from '../types/index';
 
 const PatientSearchResults: React.FC<PatientSearchResultsProps> = ({ patients, hidePanel }) => {
+  const [fhirPatients, setFhirPatients] = useState([]);
   const config = useConfig();
 
-  function renderPatient(patient: SearchedPatient) {
-    const preferredIdentifier = patient.identifiers.find((i) => i.preferred) || patient.identifiers[0];
+  useEffect(() => {
+    const fhirPatientsBuffer = [];
+    patients.forEach((patient) => {
+      fetchCurrentPatient(patient.uuid).then((response) => {
```

Comment:
This is a big red flag. What information do we need that isn't in `patients`?

Reply:
@brandones `patients` is having Patient data with OpenMRS' representation of a patient (OpenMRS custom data model - dispensed by WS/REST module) yet the patient banner expects a FHIR representation of a patient.

Reply:
I know, but what I'm asking is: what is the actual data that's needed, which isn't provided by `patients`?

Reply:
I have looked through the `PatientBanner` real quick, and I think we have all the data. I think it makes sense to manually map to fhir like objects.

Reply:
Nice! I think the best way to do it would be to define the Patient Banner interface as taking just the important parts of `patient` instead of `fhir.Patient`. You could maybe use `pick`, like

```
patient: Pick(fhir.Patient, “name” | “gender” | …
```

Reply:
Updated, also see https://github.com/openmrs/openmrs-esm-patient-chart/pull/419